benchmarks/prod.hpp

```diff
@@ -0,0 +1,24 @@
+#include "init.hpp"
+#include "benchmark.hpp"
+
+template <typename S> class prod;
+
+template <typename R, typename O1, typename O2>
```

Comment:
I would like to see different `class prod` classes for each function (matrix-matrix, matrix-vector, vector-matrix) to be benchmarked because of the initialization of the data structures.

Reply:
I'd like the benchmark name to exactly match the operation name (and likewise the signature).
Why can't you do what you have in mind (use different initialization functions) with template specializations ?

Reply:
Yes template specialization would help. But why not simplify things and put the initialization and call of the function into corresponding `mv_prod` and `mm_prod` classes. So the initialization and the execution should be within those classes. We would then have only two files with their corresponding initializations where the repsonsible functions are within one file and not distributed. `prod.hpp` and `init` becomes obsolete then.

Reply:
I expect that a few initialization functions are going to be reused across most benchmarks. And if not, specific benchmarks may pick specific initialization functions. The two are rather independent, so I chose not to bind the init functions to the benchmark classes themselves.